ykrt/src/compile/jitc_yk/codegen/x86_64/mod.rs

```diff
@@ -1259,15 +1290,24 @@ struct DeoptInfo {
     frames: Vec<u64>,
     // Vector of live JIT variable locations.
     lives: Vec<LocalAlloc>,
+    // Vector of live AOT variables.
+    aotlives: Vec<aot_ir::InstID>,
+    callframes: Vec<Frame>,
+    // Keeps track of deopt amount and compiled side-trace.
+    guard: Guard,
 }
 
 #[derive(Debug)]
 pub(super) struct X64CompiledTrace {
+    // Reference to the meta-tracer required for side tracing.
+    mt: Arc<MT>,
     /// The executable code itself.
     buf: ExecutableBuffer,
     /// Vector of deopt info, tracked here so they can be freed when the compiled trace is
     /// dropped.
     deoptinfo: Vec<DeoptInfo>,
+    /// Reference to the HotLocation, required for side tracing.
+    hl: Weak<Mutex<HotLocation>>,
```

Comment:
Maybe dumb question: Why a weak reference here?

Reply:
This is also from the old system. We must have had our reasons though I can't remember what they were. I imagine it's due to circular dependencies and we would leak otherwise.